csharp/src/Microsoft.ML.OnnxRuntime/NativeMethods.cs

```diff
@@ -564,13 +567,16 @@ static NativeMethods()
         //[DllImport(nativeLib, CharSet = charSet)]
         //public static extern void OrtAddCustomOp(IntPtr /*(OrtSessionOptions*)*/ options, string custom_op_path);
 
-        public delegate IntPtr /*(OrtStatus*)*/DOrtAddFreeDimensionOverride(IntPtr /*(OrtSessionOptions*) */ options, string /*(const char*)*/ symbolic_dim, int dim_override);
+        public delegate IntPtr /*(OrtStatus*)*/DOrtAddFreeDimensionOverride(IntPtr /*(OrtSessionOptions*) */ options, string /*(const char*)*/ dimDenotation, long dimValue);
```

Comment:
> string /*(const char*) [](start = 117, length = 23)

This is wrong. strings are not properly converted to UTF-8 and that is why we do conversion ourselves. So this must be IntPtr to a converted bytes, not string

Reply:
Thanks, I ll fix this - I think I went off other samples for this (which may be incorrect)

Reply:
There are a lot of APIs that seem to be using the default marshalling. They are all wrong then. Rather than fix piecemeal, let us do this one shot across the C# code-base in a separate change. Let us use the default marshalling for this.

Reply:
Fixed for this. Opened a work item to cover all the other APIs that need this fix across the C# code-base.